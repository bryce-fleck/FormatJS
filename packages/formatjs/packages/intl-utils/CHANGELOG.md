# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [0.4.4](https://github.com/formatjs/formatjs/compare/@formatjs/intl-utils@0.4.3...@formatjs/intl-utils@0.4.4) (2019-07-29)

**Note:** Version bump only for package @formatjs/intl-utils





## [0.4.3](https://github.com/formatjs/formatjs/compare/@formatjs/intl-utils@0.4.2...@formatjs/intl-utils@0.4.3) (2019-07-25)

**Note:** Version bump only for package @formatjs/intl-utils





## [0.4.2](https://github.com/formatjs/formatjs/compare/@formatjs/intl-utils@0.4.1...@formatjs/intl-utils@0.4.2) (2019-07-23)

**Note:** Version bump only for package @formatjs/intl-utils





## [0.4.1](https://github.com/formatjs/formatjs/compare/@formatjs/intl-utils@0.4.0...@formatjs/intl-utils@0.4.1) (2019-07-12)

**Note:** Version bump only for package @formatjs/intl-utils

# [0.4.0](https://github.com/formatjs/formatjs/compare/@formatjs/intl-utils@0.3.4...@formatjs/intl-utils@0.4.0) (2019-07-12)

### Features

- **intl-messageformat-parser:** add printer to print AST to string ([ec0eaa2](https://github.com/formatjs/formatjs/commit/ec0eaa2))

## [0.3.4](https://github.com/formatjs/formatjs/compare/@formatjs/intl-utils@0.3.3...@formatjs/intl-utils@0.3.4) (2019-07-09)

**Note:** Version bump only for package @formatjs/intl-utils

## [0.3.3](https://github.com/formatjs/formatjs/compare/@formatjs/intl-utils@0.3.2...@formatjs/intl-utils@0.3.3) (2019-07-08)

**Note:** Version bump only for package @formatjs/intl-utils

## [0.3.2](https://github.com/formatjs/formatjs/compare/@formatjs/intl-utils@0.3.1...@formatjs/intl-utils@0.3.2) (2019-06-28)

**Note:** Version bump only for package @formatjs/intl-utils

## [0.3.1](https://github.com/formatjs/formatjs/compare/@formatjs/intl-utils@0.3.0...@formatjs/intl-utils@0.3.1) (2019-07-02)

**Note:** Version bump only for package @formatjs/intl-utils

# [0.3.0](https://github.com/formatjs/formatjs/compare/@formatjs/intl-utils@0.2.1...@formatjs/intl-utils@0.3.0) (2019-06-27)

### Features

- **intl-messageformat:** allow passing in formatters ([#107](https://github.com/formatjs/formatjs/issues/107)) ([3605693](https://github.com/formatjs/formatjs/commit/3605693))

## [0.2.1](https://github.com/formatjs/formatjs/compare/@formatjs/intl-utils@0.2.0...@formatjs/intl-utils@0.2.1) (2019-06-26)

**Note:** Version bump only for package @formatjs/intl-utils

# 0.2.0 (2019-06-27)

### Features

- **intl-utils:** Add intl-utils ([#98](https://github.com/formatjs/formatjs/issues/98)) ([2329c57](https://github.com/formatjs/formatjs/commit/2329c57))
