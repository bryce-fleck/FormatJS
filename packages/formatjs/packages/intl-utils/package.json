{
  "name": "@formatjs/intl-utils",
  "version": "0.5.0",
  "description": "Smartly determine best unit for relative time format",
  "keywords": [
    "intl",
    "i18n",
    "relative",
    "moment",
    "format",
    "date"
  ],
  "author": "Long Ho <holevietlong@gmail.com",
  "bugs": {
    "url": "https://github.com/formatjs/formatjs/issues"
  },
  "repository": {
    "type": "git",
    "url": "git@github.com:formatjs/formatjs.git"
  },
  "main": "dist/index",
  "types": "dist/index.d.ts",
  "type": "module",
  "dependencies": {
    "@formatjs/intl-relativetimeformat": "^2.5.1",
    "date-fns": "^1.30.1"
  },
  "scripts": {
    "mjs": "find ./dist -depth -name \"*.js\" -exec sh -c 'f=\"{}\"; mv -- \"$f\" \"${f%.js}.mjs\"' \\;",
    "clean": "rimraf dist lib",
    "build": "tsc && npm run mjs && tsc -p tsconfig.cjs.json",
    "test": "mocha --opts ../../mocha.opts tests/index.ts"
  },
  "homepage": "https://github.com/formatjs/formatjs",
  "license": "MIT",
  "gitHead": "8b0baec8eda5002715cf893274fe59782fc2d371"
}
