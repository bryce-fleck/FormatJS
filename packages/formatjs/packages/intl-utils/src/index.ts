export {selectUnit} from './diff';
export {resolveSupportedLocales} from './locale-lookup';
