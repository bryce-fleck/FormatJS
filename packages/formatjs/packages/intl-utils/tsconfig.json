{
    "extends": "../../tsconfig.json",
    "compilerOptions": {
        "outDir": "dist",
        "rootDir": "src",
        "baseUrl": ".",
        "paths": {
            "@formatjs/intl-relativetimeformat": ["../intl-relativetimeformat"]
        }
    },
    "include": ["src/*.ts"]
}