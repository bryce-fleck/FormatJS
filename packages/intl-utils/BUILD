load("@build_bazel_rules_nodejs//:index.bzl", "pkg_npm")
load("@npm//@bazel/typescript:index.bzl", "ts_project")
load("//tools:index.bzl", "rollup_dts")
load("//tools:jest.bzl", "jest_test")

PACKAGE_NAME = "intl-utils"

pkg_npm(
    name = PACKAGE_NAME,
    package_name = "@formatjs/%s" % PACKAGE_NAME,
    srcs = [
        "LICENSE.md",
        "package.json",
    ],
    deps = [
        ":dist",
        ":lib",
    ],
)

SRCS = glob(["src/*.ts"]) + ["index.ts"]

SRC_DEPS = []

ts_project(
    name = "dist",
    package_name = "@formatjs/%s" % PACKAGE_NAME,
    srcs = SRCS,
    declaration = True,
    declaration_map = True,
    source_map = True,
    tsconfig = "//:tsconfig.json",
    visibility = ["//visibility:public"],
    deps = SRC_DEPS,
)

ts_project(
    name = "lib",
    srcs = SRCS,
    declaration = True,
    declaration_map = True,
    extends = ["//:tsconfig.json"],
    outdir = "lib",
    source_map = True,
    tsconfig = "//:tsconfig.esm.json",
    deps = SRC_DEPS,
)

filegroup(
    name = "types",
    srcs = [":dist"],
    output_group = "types",
    visibility = ["//visibility:public"],
)

jest_test(
    name = "unit",
    srcs = SRCS + glob([
        "tests/**/*.ts",
        "tests/**/*.tsx",
        "tests/**/*.snap",
    ]),
    deps = SRC_DEPS,
)

rollup_dts(
    name = "rollup-dts",
    package_name = PACKAGE_NAME,
    package_json = "package.json",
    types = [":types"],
)
