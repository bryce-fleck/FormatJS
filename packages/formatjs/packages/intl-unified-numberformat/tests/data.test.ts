import {
  extractStandardCurrencyPattern,
  replaceCurrencySymbolWithToken,
} from '../src/data';
import {InternalSlotToken} from '@formatjs/intl-utils';
describe('data', function() {
  it('extractStandardCurrencyPattern', function() {
    expect(
      extractStandardCurrencyPattern(
        '¤#,##0.00',
        '{0} {1}',
        '¤0K',
        '0K',
        '0 thousand',
        'narrowSymbol'
      )
    ).toMatchSnapshot();
  });
  it('replaceCurrencySymbolWithToken', function() {
    expect(
      replaceCurrencySymbolWithToken(
        'ZWD',
        '{plusSign}¤{number}{compactSymbol}',
        ' ',
        InternalSlotToken.currencyNarrowSymbol
      )
    ).toMatchInlineSnapshot(
      `"{plusSign}{currencyNarrowSymbol} {number}{compactSymbol}"`
    );
  });
});
