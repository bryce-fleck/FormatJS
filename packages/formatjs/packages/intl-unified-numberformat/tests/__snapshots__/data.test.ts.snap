// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`data extractStandardCurrencyPattern 1`] = `
Object {
  "always": Object {
    "compactLong": Object {
      "negativePattern": "{minusSign}¤{number}{compactSymbol}",
      "positivePattern": "{plusSign}¤{number}{compactSymbol}",
      "zeroPattern": "{plusSign}¤{number}{compactSymbol}",
    },
    "compactShort": Object {
      "negativePattern": "{minusSign}¤{number}{compactSymbol}",
      "positivePattern": "{plusSign}¤{number}{compactSymbol}",
      "zeroPattern": "{plusSign}¤{number}{compactSymbol}",
    },
    "scientific": Object {
      "negativePattern": "{minusSign}¤{number}{scientificSeparator}{scientificExponent}",
      "positivePattern": "{plusSign}¤{number}{scientificSeparator}{scientificExponent}",
      "zeroPattern": "{plusSign}¤{number}{scientificSeparator}{scientificExponent}",
    },
    "standard": Object {
      "negativePattern": "{minusSign}¤{number}",
      "positivePattern": "{plusSign}¤{number}",
      "zeroPattern": "{plusSign}¤{number}",
    },
  },
  "auto": Object {
    "compactLong": Object {
      "negativePattern": "{minusSign}¤{number}{compactSymbol}",
      "positivePattern": "¤{number}{compactSymbol}",
      "zeroPattern": "¤{number}{compactSymbol}",
    },
    "compactShort": Object {
      "negativePattern": "{minusSign}¤{number}{compactSymbol}",
      "positivePattern": "¤{number}{compactSymbol}",
      "zeroPattern": "¤{number}{compactSymbol}",
    },
    "scientific": Object {
      "negativePattern": "{minusSign}¤{number}{scientificSeparator}{scientificExponent}",
      "positivePattern": "¤{number}{scientificSeparator}{scientificExponent}",
      "zeroPattern": "¤{number}{scientificSeparator}{scientificExponent}",
    },
    "standard": Object {
      "negativePattern": "{minusSign}¤{number}",
      "positivePattern": "¤{number}",
      "zeroPattern": "¤{number}",
    },
  },
  "exceptZero": Object {
    "compactLong": Object {
      "negativePattern": "{minusSign}¤{number}{compactSymbol}",
      "positivePattern": "{plusSign}¤{number}{compactSymbol}",
      "zeroPattern": "¤{number}{compactSymbol}",
    },
    "compactShort": Object {
      "negativePattern": "{minusSign}¤{number}{compactSymbol}",
      "positivePattern": "{plusSign}¤{number}{compactSymbol}",
      "zeroPattern": "¤{number}{compactSymbol}",
    },
    "scientific": Object {
      "negativePattern": "{minusSign}¤{number}{scientificSeparator}{scientificExponent}",
      "positivePattern": "{plusSign}¤{number}{scientificSeparator}{scientificExponent}",
      "zeroPattern": "¤{number}{scientificSeparator}{scientificExponent}",
    },
    "standard": Object {
      "negativePattern": "{minusSign}¤{number}",
      "positivePattern": "{plusSign}¤{number}",
      "zeroPattern": "¤{number}",
    },
  },
  "never": Object {
    "compactLong": Object {
      "negativePattern": "¤{number}{compactSymbol}",
      "positivePattern": "¤{number}{compactSymbol}",
      "zeroPattern": "¤{number}{compactSymbol}",
    },
    "compactShort": Object {
      "negativePattern": "¤{number}{compactSymbol}",
      "positivePattern": "¤{number}{compactSymbol}",
      "zeroPattern": "¤{number}{compactSymbol}",
    },
    "scientific": Object {
      "negativePattern": "¤{number}{scientificSeparator}{scientificExponent}",
      "positivePattern": "¤{number}{scientificSeparator}{scientificExponent}",
      "zeroPattern": "¤{number}{scientificSeparator}{scientificExponent}",
    },
    "standard": Object {
      "negativePattern": "¤{number}",
      "positivePattern": "¤{number}",
      "zeroPattern": "¤{number}",
    },
  },
}
`;

exports[`data replaceCurrencySymbolWithToken 1`] = `"{plusSign}{currencyNarrowSymbol} {number}{compactSymbol}"`;
